File: php/TennisGame2.php
<?php


const pointScores = array('Love', 'Fifteen', 'Thirty', 'Forty');

class TennisGame2 implements TennisGame
{
    private $P1point = 0;
    private $P2point = 0;

    private $P1res = "";
    private $P2res = "";
    private $player1Name = "";
    private $player2Name = "";

    public function __construct($player1Name, $player2Name)
    {
        $this->player1Name = $player1Name;
        $this->player2Name = $player2Name;
    }

    public function getScore()
    {
        $equality = $this->P1point == $this->P2point;
        if ($equality) {
            $beforeDeuce = $this->P1point < 3;
            if ($beforeDeuce) {
                $score = pointScores[$this->P1point]."-All";
            } else {
                $score = "Deuce";
            }
        } else {

            $leader = $this->P1point > $this->P2point ? $this->player1Name : $this->player2Name;

            if (max($this->P1point, $this->P2point) >= 4 && abs($this->P1point - $this->P2point) >= 2) {
                $score = "Win for $leader";
            } else if (min($this->P1point, $this->P2point) >= 3) {
                $score = "Advantage $leader";
            } else  {
                $this->P1res = pointScores[$this->P1point];
                $this->P2res = pointScores[$this->P2point];
                $score = "{$this->P1res}-{$this->P2res}";
            }

        }

        return $score;
    }

    private function SetP1Score($number)
    {
        for ($i = 0; $i < $number; $i++) {
            $this->P1Score();
        }
    }

    private function SetP2Score($number)
    {
        for ($i = 0; $i < $number; $i++) {
            $this->P2Score();
        }
    }

    private function P1Score()
    {
        $this->P1point++;
    }

    private function P2Score()
    {
        $this->P2point++;
    }

    public function wonPoint($player)
    {
        if ($player == "player1") {
            $this->P1Score();
        } else {
            $this->P2Score();
        }
    }
}

File: java/src/main/java/TennisGame2.java
import java.util.Arrays;
import java.util.List;

public class TennisGame2 implements TennisGame {
    public int P1point = 0;
    public int P2point = 0;

    public String P1res = "";
    public String P2res = "";
    private String player1Name;
    private String player2Name;

    private List<String> ballPoints = Arrays.asList("Love", "Fifteen", "Thirty", "Forty");

    public TennisGame2(String player1Name, String player2Name) {
        this.player1Name = player1Name;
        this.player2Name = player2Name;
    }

    public String getScore() {
        String score = "";
        if (scoreIsEqual() && P1point >= 3) {

            score = "Deuce";

        } else if (scoreIsEqual()) {

            score = getPointScore(P1point) + "-All";

        } else if (gameNotFinished()) {

            score = getPointScore(P1point) + "-" + getPointScore(P2point);

        } else if (isAdvantage()) {

            score = "Advantage " + getLeaderName();

        } else {

            score = "Win for " + getLeaderName();
        }

        return score;
    }

    private boolean scoreIsEqual() {
        return P1point == P2point;
    }

    private boolean isAdvantage() {
        return Math.abs(P1point - P2point) == 1;
    }

    private boolean gameNotFinished() {
        return P1point < 4 && P2point < 4;
    }

    private String getPointScore(int p1point) {
        return ballPoints.get(p1point);
    }

    private String getLeaderName() {
        return P1point > P2point ? "player1" : "player2";
    }

    public void SetP1Score(int number) {

        for (int i = 0; i < number; i++) {
            P1Score();
        }

    }

    public void SetP2Score(int number) {

        for (int i = 0; i < number; i++) {
            P2Score();
        }

    }

    public void P1Score() {
        P1point++;
    }

    public void P2Score() {
        P2point++;
    }

    public void wonPoint(String player) {
        if (player == "player1")
            P1Score();
        else
            P2Score();
    }
}